core/src/main/scala/chisel3/Layer.scala

```diff
@@ -19,35 +19,48 @@ object layer {
   /** Enumerations of different optional layer conventions.  A layer convention
     * says how a given layer should be lowered to Verilog.
     */
+  @deprecated
```

Comment:
The `@deprecated` annotation takes two optional arguments:

1. A string suggesting what the user should do instead.
2. A string indicating since what version this is deprecated.

Can the PR include something useful here that is actionable by a user?

```suggestion
  @deprecated("`Convention` is being removed in favor of `LayerConfig`", "Chisel 7.0.0")
```

Same comment for the deprecated auxiliary constructor.